CRM/Contribute/Form/ContributionBase.php

```diff
@@ -313,6 +313,17 @@ public function preProcess() {
         throw new CRM_Contribute_Exception_InactiveContributionPageException(ts('The page you requested is currently unavailable.'), $this->_id);
       }
 
+      $endDate = CRM_Utils_Date::processDate(CRM_Utils_Array::value('end_date', $this->_values));
+      $now = date('YmdHis');
+      if ($endDate && $endDate < $now) {
+        throw new Exception(ts('The page you requested has past its end date on '. CRM_Utils_Date::customFormat($endDate) ), $this->_id);
```

Comment:
should we be chucking Exception or `CRM_Contribute_Exception_InactiveContributionPageException` exception? @eileenmcnaughton

Reply:
ohh clever spotting!